src/System.Runtime.Numerics/tests/BigInteger/parse.cs

```diff
@@ -321,167 +320,129 @@ public static void VerifyNumberStyles(NumberStyles ns, Random random)
             }
 
             // Invalid Chars
             for (int i = 0; i < s_samples; i++)
             {
-                Assert.True(VerifyParseToString(GetDigitSequence(1, 100, random) + GetRandomInvalidChar(random) + GetDigitSequence(1, 10, random), ns, false), " Verification Failed");
+                VerifyParseToString(GetDigitSequence(1, 100, random) + GetRandomInvalidChar(random) + GetDigitSequence(1, 10, random), ns, false);
             }
         }
 
-        private static bool VerifyParseToString(string num1)
+        private static void VerifyParseToString(string num1)
         {
-            bool ret = true;
             BigInteger test;
 
-            ret &= Eval(BigInteger.Parse(num1), Fix(num1.Trim()), String.Format("Roundtrip between Parse() and ToString() failed.  RoundTrip returned: {0}  num1: {1}", BigInteger.Parse(num1), Fix(num1.Trim())));
-            ret &= BigInteger.TryParse(num1, out test);
-            ret &= Eval(test, Fix(num1.Trim()), String.Format("Roundtrip between TryParse() and ToString() failed.  ToString returned: {0} num1: {1}", test, Fix(num1.Trim())));
-
-            return ret;
+            Eval(BigInteger.Parse(num1), Fix(num1.Trim()));
+            Assert.True(BigInteger.TryParse(num1, out test));
+            Eval(test, Fix(num1.Trim()));
         }
-        private static bool VerifyFailParseToString(string num1, Type expect)
-        {
-            bool ret = true;
-            BigInteger test;
 
-            try
+        private static void VerifyFailParseToString(string num1, Type expectedExceptionType)
+        {
+            if (expectedExceptionType == new FormatException().GetType())
             {
-                Eval(BigInteger.Parse(num1).ToString("d"), num1.Trim(), String.Format("Roundtrip between Parse() and ToString() failed.  RoundTrip returned: {0}  num1: {1}", BigInteger.Parse(num1).ToString("d"), num1.Trim()));
-                Console.WriteLine("{0} did not throw expected exception", num1);
-                ret = false;
+                Assert.Throws<FormatException>(() => { BigInteger.Parse(num1).ToString("d"); });
             }
-            catch (Exception e)
+            else if (expectedExceptionType == new ArgumentNullException().GetType())
             {
-                if (!(e.GetType() == expect))
-                {
-                    Console.WriteLine("Wrong exception thrown.\n" + e);
-                    ret = false;
-                }
+                Assert.Throws<ArgumentNullException>(() => { BigInteger.Parse(num1).ToString("d"); });
+            }
+            else
+            {
+                Assert.True(false, "Unsupported exception type expected");
             }
-            ret &= Eval(!BigInteger.TryParse(num1, out test), true, String.Format("TryParse expected on fail on {0}", num1));
 
-            return ret;
+            BigInteger test;
```

Comment:
What makes you say that? Calls that pass either `ArgumentNullException` or `FormatException` for `expectedExceptionType` and that fail on `Parse()` will continue to `TryParse()`, and I verified with a breakpoint.